tiledb/libtiledb.pyx

```diff
@@ -3769,24 +3769,51 @@ cdef ArrayPtr preload_array(uri, mode, key, timestamp, Ctx ctx=None):
         key_ptr = <void *> PyBytes_AS_STRING(bkey)
         #TODO: unsafe cast here ssize_t -> uint64_t
         key_len = <unsigned int> PyBytes_GET_SIZE(bkey)
-    cdef uint64_t _timestamp = 0
+    cdef uint64_t ts_start = 0
+    cdef uint64_t ts_end = 0
+    cdef bint set_start = False, set_end = False
+
     if timestamp is not None:
-        _timestamp = <uint64_t> timestamp
+        if isinstance(timestamp, tuple):
+            if len(timestamp) != 2:
+                raise ValueError("'timestamp' argument expects either int or tuple(start: int, end: int)")
+            if timestamp[0] is not None:
+                ts_start = <uint64_t>timestamp[0]
+                set_start = True
+            if timestamp[1] is not None:
+                ts_end = <uint64_t>timestamp[1]
+                set_end = True
+        elif isinstance(timestamp, int):
+            ts_end = <uint64_t> timestamp
+            set_start = True
```

Comment:
Should this be false?

Reply:
Cruft, thanks for the catch. Will remove.